src/cephadm/cephadm

```diff
@@ -2060,7 +2060,7 @@ KillMode=none
 Restart=on-failure
 RestartSec=10s
 TimeoutStartSec=120
-TimeoutStopSec=15
+TimeoutStopSec=120
```

Comment:
Two minutes is super long.

My random thoughts are:

* Processes need to cope with getting killed anyway. And if your application takes ages to terminate, users will kill it much more often.
* This might slow down cephadm considerably, making cephadm less responsive. Which is already a big issue.
* This might lead to problems, if users think a process is already restarted, like new configuration values that are getting applied only after a restart.

**can we increase the timeout only for ganesha-grace?**

Reply:
> Two minutes is super long.

Most (all?) services should shutdown well before the 2min timeout has been reached.


> **can we increase the timeout only for ganesha-grace?**

OSD shutdown suffers from the same issue.

In this case `ceph-volume lvm deactivate` from the unit.poststop script never occurred:
```
May 07 18:04:31 host1 systemd[1]: Stopping Ceph osd.1 for 00000000-0000-0000-0000-0000deadbeef...
May 07 18:04:31 host1 bash[5244]: debug 2020-05-08T00:04:31.590+0000 7fb38b4cb700 -1 received  signal: Terminated from Kernel ( Could be generated by pthread_kill(), raise(), abort(), alarm() ) UID: 0
May 07 18:04:31 host1 bash[5244]: debug 2020-05-08T00:04:31.590+0000 7fb38b4cb700 -1 osd.1 17 *** Got signal Terminated ***
May 07 18:04:31 host1 bash[5244]: debug 2020-05-08T00:04:31.590+0000 7fb38b4cb700 -1 osd.1 17 *** Immediate shutdown (osd_fast_shutdown=true) ***
May 07 18:04:35 host1 podman[20537]: 2020-05-07 18:04:35.042806814 -0600 MDT m=+3.565533584 container died c4e8483cc3fa2cb5be45d30835673bc8ca9b37a503ba4dad81a699f9338423cf (image=docker.io/ceph/daemon-base:latest-master-devel, name=ceph-00000000-0000-0000-0000-0000deadbeef-osd.1)
May 07 18:04:36 host1 podman[20537]: 2020-05-07 18:04:36.568134109 -0600 MDT m=+5.090860868 container stop c4e8483cc3fa2cb5be45d30835673bc8ca9b37a503ba4dad81a699f9338423cf (image=docker.io/ceph/daemon-base:latest-master-devel, name=ceph-00000000-0000-0000-0000-0000deadbeef-osd.1)
May 07 18:04:36 host1 podman[20537]: c4e8483cc3fa2cb5be45d30835673bc8ca9b37a503ba4dad81a699f9338423cf
May 07 18:04:47 host1 podman[5244]: 2020-05-07 18:04:47.556767081 -0600 MDT m=+20088.059495793 container remove c4e8483cc3fa2cb5be45d30835673bc8ca9b37a503ba4dad81a699f9338423cf (image=docker.io/ceph/daemon-base:latest-master-devel, name=ceph-00000000-0000-0000-0000-0000deadbeef-osd.1)
May 07 18:04:51 host1 systemd[1]: ceph-00000000-0000-0000-0000-0000deadbeef@osd.1.service: State 'stop-post' timed out. Terminating.
May 07 18:04:51 host1 systemd[1]: ceph-00000000-0000-0000-0000-0000deadbeef@osd.1.service: Failed with result 'timeout'.
May 07 18:04:51 host1 systemd[1]: Stopped Ceph osd.1 for 00000000-0000-0000-0000-0000deadbeef.
```